packages/deribit/lib/index.ts

```diff
@@ -0,0 +1,65 @@
+import { OptionType } from '@node-dlc/core';
+
+export const getStrDate = (timestamp: number): string => {
+  const date = new Date(timestamp * 1000);
+
+  const day = date.getUTCDate();
+  const month = date
+    .toLocaleString('default', { month: 'short', timeZone: 'UTC' })
+    .toUpperCase()
+    .replace('.', '');
+  const year = date.toLocaleDateString('pl', {
+    year: '2-digit',
+    timeZone: 'UTC',
+  });
+
+  return `${day}${month}${year}`;
+};
+
+type OptionInstrument = {
+  expiry: number;
+  strikePrice: number;
+  type: OptionType;
+};
+
+export const composeInstrumentName = ({
+  expiry,
+  type,
+  strikePrice,
+}: OptionInstrument): string => {
+  const strDate = getStrDate(expiry);
+  const typeSymbol = type === 'call' ? 'C' : 'P';
+  return `BTC-${strDate}-${strikePrice}-${typeSymbol}`;
+};
+
+const EXPIRY_REGEX = /(\d{1,2})([A-Z]+)(\d{1,2})/;
+export const parseInstrumentName = (
+  instrumentName: string,
+): OptionInstrument => {
+  try {
+    const [, expiryString, _strikePrice, typeSymbol] =
+      instrumentName.split('-');
+    if (
+      !typeSymbol ||
+      !EXPIRY_REGEX.test(expiryString) ||
+      (typeSymbol !== 'C' && typeSymbol !== 'P')
+    ) {
+      throw new Error(`Only option instrument names are supported.`);
```

Comment:
would be great to have a test case for this as well

Reply:
This is already tested in https://github.com/AtomicFinance/atomic-utils/pull/3/files#diff-28d9efc8d642c55b60fe7d88015737bc9fa2eacda8f6e203759e3881a38db6a7R108

The error thrown is rethrown above with a different message, since there's a general catch all around the function.